src/app/app.component.html

```diff
@@ -6,7 +6,13 @@
     (onActionSelect)="handleAction($event)">
 </pfng-toast-notification-list>
 <f8-add-space-overlay class="launcher fade in" *ngIf="showAddSpaceOverlay"></f8-add-space-overlay>
-<f8-add-app-overlay [preselectedFlow]="addAppFlow" class="code-imports fade in" *ngIf="showAddAppOverlay"></f8-add-app-overlay>
+<div class="modal fade" bsModal #modalAddAppOverlay="bs-modal" tabindex="-1" role="dialog" aria-hidden="true">
```

Comment:
This functionality should be placed in a separate component. The app component is being overloaded with extra concerns that it shouldn't need to worry about.
Same could be said about the `f8-add-space-overlay`. The controlling of opening the overlay shouldn't be the concern of the app but rather the overlay component, or a wrapper thereof.

The modal markup should be placed in a separate component such as `add-app-overlay-modal`.

That being said, since the existing functionality was in the app. I'm ok for now to accept this and track refactoring both in another issue.

Reply:
I completely agree. The app component is the entry point and should just be calling the highest level component or components.

You have added a `<div>` with a modal to the highest parent in the UI. This should be fixed before it is merged. This could introduce unforeseen problems.

Reply:
Done in https://github.com/fabric8-ui/fabric8-ui/pull/3427/commits/bb3764fe8083dcddf92b231d1c71f8cac451e981